front/src/common/Map/Layers/InfraObjectLayers/SpeedLimits.tsx

```diff
@@ -15,20 +16,28 @@ interface SpeedLimitsProps {
   layerOrder: number;
   punctualLayerOrder: number;
   infraID?: number;
-  layersSettings: LayersSettings;
+  layersSettings: MapState['mapSettings']['layersSettings'];
+  speedlimittag?: string;
   highlightedArea?: Geometry;
 }
 
-export function getSpeedSectionsTag({ speedlimittag }: LayersSettings): string {
-  return speedlimittag !== null ? `speed_limit_by_tag_${speedlimittag}` : 'null';
+export function getSpeedSectionsTag(
+  layersSettings?: LayersSettings,
+  speedlimittag?: string
+): string {
```

Comment:
Hm it would be better to always give in argument a speedLimitTag
```suggestion
export function getSpeedSectionsTag(speedlimittag?: string): string {
```

Then, you can do `getSpeedSectionsTag(speedlimittag ?? layersSettings?.speedlimittag)` if you need when you call the function